vmm/src/lib.rs

```diff
@@ -229,6 +230,42 @@ impl std::convert::From<DriveError> for VmmActionError {
     }
 }
 
+// It's convenient to turn VmConfigErrors into VmmActionErrors directly.
+impl std::convert::From<VmConfigError> for VmmActionError {
+    fn from(e: VmConfigError) -> Self {
+        VmmActionError::MachineConfig(
+            match e {
+                VmConfigError::InvalidVcpuCount
+                | VmConfigError::InvalidMemorySize
+                | VmConfigError::UpdateNotAllowedPostBoot => ErrorKind::User,
+            },
+            e,
+        )
+    }
+}
+
+// It's convenient to turn NetworkInterfaceErrors into VmmActionErrors directly.
+impl std::convert::From<NetworkInterfaceError> for VmmActionError {
+    fn from(e: NetworkInterfaceError) -> Self {
+        let kind = match e {
+            NetworkInterfaceError::GuestMacAddressInUse(_) => ErrorKind::User,
```

Comment:
Same observation with grouping error kinds according to what they match to.

Reply:
Sounds good.